tests/test_lsst.py

```
@@ -0,0 +1,928 @@
+             pickle.dump(self.wcs, output_file)
+
+        with open(file_name, 'r') as input_file:
+            wcs1 = pickle.load(input_file)
+
+        self.assertEqual(self.wcs, wcs1)
+
+        if os.path.exists(file_name):
+            os.unlink(file_name)
+
+
+    def test_passing_camera_by_hand(self):
+        """
+        Test that you can pass a camera from one WCS to another
+        """
+
+        with warnings.catch_warnings(record=True) as ww:
+            wcs1 = LsstWCS(self.pointing, self.rotation, chip_name='R:0,1 S:1,1', camera=self.wcs.camera)
+
+        self.assertEqual(len(ww), 0)
+
+
+        # verify that, if the camera does not have the pointing or rotation angle you want,
+        # a new camera will be instantiated
+        with warnings.catch_warnings(record=True) as ww:
+            wcs1 = LsstWCS(galsim.CelestialCoord(0.0*galsim.degrees, 0.0*galsim.degrees), self.rotation,
+                           chip_name='R:0,1 S:1,1', camera=self.wcs.camera)
+
+        self.assertEqual(str(ww[0].message),
+                         "The camera you passed to LsstWCS does not have the same\n"
+                         "pointing and rotation angle as you asked for for this WCS.\n"
+                         "LsstWCS is creating a new camera with the pointing and\n"
+                         "rotation angle you specified in the constructor for LsstWCS.")
+
+
+        with warnings.catch_warnings(record=True) as ww:
+            wcs1 = LsstWCS(self.pointing, 49.0*galsim.degrees,
+                           chip_name='R:0,1 S:1,1', camera=self.wcs.camera)
+
+        self.assertEqual(str(ww[0].message),
+                         "The camera you passed to LsstWCS does not have the same\n"
+                         "pointing and rotation angle as you asked for for this WCS.\n"
+                         "LsstWCS is creating a new camera with the pointing and\n"
+                         "rotation angle you specified in the constructor for LsstWCS.")
```

Regarding testing the exact wording of the warning: I like to do that to make sure that the warning (or exception) being raised is the one I thought I was provoking in the unit test (rather than some other warning from earlier in the traceback).  I have been told this is excessively paranoid and fragile.  If you would prefer I not do that, I can just test that a warning is raised.